### ISSUE TITLE
ArbitraryIntrospector not working as expected

### ISSUE THREAD

**Issue by esfomeado** [naver/fixture-monkey#595]
## Describe the bug

When configuring Fixture Monkey with:

```
.pushAssignableTypeArbitraryIntrospector(Record.class, ConstructorPropertiesArbitraryIntrospector.INSTANCE)
```
I would expect all Records to use the `ConstructorPropertiesArbitaryIntrospector` but instead if the Record implements an interface it uses the default introspector.

## Your environment
* Fixture Monkey: 0.5.1
* Java 17

## Steps to reproduce
https://github.com/esfomeado/fixture-monkey-bug/blob/590c0d69522ce0fd2adbc725b29be1992efee858/src/test/java/com/example/bug/FixtureMonkeyTest.java

If I add this, it works fine but I would expect this not to be needed.
` .pushAssignableTypeArbitraryIntrospector(Book.class, ConstructorPropertiesArbitraryIntrospector.INSTANCE)`

## Expected behaviour
Should generate the class using the `ConstructorPropertiesArbitaryIntrospector`

## Actual behaviour
It´s using the `FieldReflectionArbitraryIntrospector`

**Comment by seongahjo** [comment]
@esfomeado 
Concrete class `FantasyBook` is an implementation of `Record`, but `Book` is an interface not an implementation of `Record`, the option is not work.

It is work if you generates `FantasyBook` just as below.

```java
@Test
    void testMe() {
        Book book = fixtureMonkey.giveMeOne(FantasyBook.class);

        assertThat(book).isNotNull();
    }
```

I think additional option is needed 
`pushAssignableTypeArbitraryIntrospector(Book.class, ConstructorPropertiesArbitraryIntrospector.INSTANCE)`

**Comment by seongahjo** [comment]
@esfomeado 
It fixed in 0.5.2.
Please check it out 😄

**Comment by esfomeado** [comment]
Thanks it works perfectly now.
